### PR TITLE
[new release] tyxml-ppx, tyxml-syntax, tyxml and tyxml-jsx (4.5.0)

### PR DESCRIPTION
**Pull request by Drup**
CHANGES:

* Move all the PPXs to ppxlib
  (ocsigen/tyxml#271, Initial code by Sonja @pitag-ha Heinze)

* Add the `translate` attribute
  (ocsigen/tyxml#281 by Javier Chávarri)
* Update allowed `inputmode`s
  (ocsigen/tyxml#279 by Joel Burget)
* Add the `picture` element
  (ocsigen/tyxml#263 by Stéphane Legrand)

### CODE REVIEW

File: packages/tyxml-jsx/tyxml-jsx.4.5.0/opam [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+maintainer: ["dev@ocsigen.org"]
+authors: ["The ocsigen team"]
+license: "LGPL-2.1 with OCaml linking exception"
+homepage: "https://github.com/ocsigen/tyxml"
+doc: "https://ocsigen.org/tyxml/latest/manual/intro"
+bug-reports: "https://github.com/ocsigen/tyxml/issues"
+depends: [
+  "dune" {>= "2.0"}
+  "ocaml" {>= "4.04"}
+  "tyxml" {version}
```
Comment (kit-ty-kate) [inline packages/tyxml-jsx/tyxml-jsx.4.5.0/opam:21]:
```suggestion
  "tyxml" {= version}
```

File: packages/tyxml-jsx/tyxml-jsx.4.5.0/opam [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  "tyxml-syntax" {version}
+  "alcotest" {with-test}
+  "reason" {with-test}
+  "ppxlib"
+]
+build: [
+  ["dune" "subst"] {pinned}
```
Comment (kit-ty-kate) [inline packages/tyxml-jsx/tyxml-jsx.4.5.0/opam:28]:
```suggestion
  ["dune" "subst"] {dev}
```

File: packages/tyxml-ppx/tyxml-ppx.4.5.0/opam [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+doc: "https://ocsigen.org/tyxml/latest/manual/intro"
+bug-reports: "https://github.com/ocsigen/tyxml/issues"
+depends: [
+  "dune" {>= "2.0"}
+  "ocaml" {>= "4.04"}
+  "tyxml" {version}
+  "tyxml-syntax" {version}
+  "alcotest" {with-test}
+  "markup" {>= "0.7.2"}
+  "ppxlib"
```
Comment (kit-ty-kate) [inline packages/tyxml-ppx/tyxml-ppx.4.5.0/opam:25]: I highly suspect this requires a lower-bound constraint (at least >= 0.9 or something like that)

**Review by kit-ty-kate — CHANGES_REQUESTED** [review]

File: packages/tyxml-jsx/tyxml-jsx.4.5.0/opam [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+open Tyxml;
+let to_reason = <a href="reasonml.github.io/"> "Reason!" </a>
+```
+ 
+The TyXML JSX allow to write TyXML documents with reason's JSX syntax. 
+It works with textual trees, virtual DOM trees, or any TyXML module.
+"""
+maintainer: ["dev@ocsigen.org"]
+authors: ["The ocsigen team"]
+license: "LGPL-2.1 with OCaml linking exception"
```
Comment (kit-ty-kate) [inline packages/tyxml-jsx/tyxml-jsx.4.5.0/opam:14]:
```suggestion
license: "LGPL-2.1-only with OCaml-LGPL-linking-exception"
```
to be SPDX-compatible

File: packages/tyxml-jsx/tyxml-jsx.4.5.0/opam (hunk 2 above)
Comment (Drup) [reply packages/tyxml-jsx/tyxml-jsx.4.5.0/opam]: cc @rgrinberg , this is generated by dune.

**Comment by Drup** [comment]
@kit-ty-kate Thanks! All done.

File: packages/tyxml-jsx/tyxml-jsx.4.5.0/opam (hunk 2 above)
Comment (kit-ty-kate) [reply packages/tyxml-jsx/tyxml-jsx.4.5.0/opam]: yes, this as been fixed in dune 2.7. opam file generation in dune just requires you to upgrade to `(lang dune 2.7)` to generated the fixed version

File: packages/tyxml-jsx/tyxml-jsx.4.5.0/opam [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+doc: "https://ocsigen.org/tyxml/latest/manual/intro"
+bug-reports: "https://github.com/ocsigen/tyxml/issues"
+depends: [
+  "dune" {>= "2.0"}
+  "ocaml" {>= "4.04"}
+  "tyxml" {= version}
+  "tyxml-syntax" {= version}
+  "alcotest" {with-test}
+  "reason" {with-test}
+  "ppxlib" {>= "0.9"}
```
Comment (kit-ty-kate) [inline packages/tyxml-jsx/tyxml-jsx.4.5.0/opam:25]: We'll have a lower-bound constraint checker in the CI in a week or two but in the meantime, looking at the code, I believe 0.9 is not high enough. Could try what fits? (try 0.18 first)

Comment (Drup) [reply packages/tyxml-jsx/tyxml-jsx.4.5.0/opam]: Ok, I tried, and 0.18 seems to be indeed the minimal version. Thanks a lot.

**Comment by kit-ty-kate** [comment]
Thanks!

**Pull request merged**